components/slot-fill/slot.js

```diff
@@ -30,25 +36,40 @@ class Slot extends Component {
 
 		if ( this.props.name !== name ) {
 			unregisterSlot( this.props.name );
-			registerSlot( name, this.node );
+			registerSlot( name, this );
 		}
 	}
 
-	registerSlot( node ) {
-		const { registerSlot = noop } = this.context;
-
+	bindNode( node ) {
 		this.node = node;
-		registerSlot( this.props.name, node );
 	}
 
 	render() {
-		return <div ref={ this.registerSlot } />;
+		const { name, bubble = false } = this.props;
+		const { getFills = noop } = this.context;
+
+		if ( bubble ) {
+			return <div ref={ this.bindNode } />;
+		}
+
+		return (
+			<div ref={ this.bindNode }>
+				{ map( getFills( name ), ( fill ) => {
+					const fillKey = fill.props.instanceId;
+					return Children.map( fill.props.children, ( child, childIndex ) => {
+						const childKey = fillKey + '---' + child.props.key || childIndex;
```

Comment:
I don't think the behavior here with `||` is as you'd expect:

```js
'foo' + '---' + null || 100
// "foo---null"
```

Order of operations means the `||` compares against the concatenated string, not `key`. I think you want:

```js
'foo' + '---' + ( null || 100 )
// "foo---100"
```

Reply:
For what reason do we need the `'---'` ?

Reply:
It's just a random separator to limit issues like `ab`+ `c`= `a` + `bc`. But yeah I need parenthesis.